### PR TITLE
Add QMK support for Okayu

### PR DESCRIPTION
**Pull request by MaiTheSan**
## Description
Add support for Salane Okayu PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sawnsprojects/okayu/readme.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Okayu
+
+![Okayu](https://i.imgur.com/pRnRHl8.png)
```
Comment (drashna) [inline keyboards/sawnsprojects/okayu/readme.md:3]:
```suggestion
![Okayu](https://i.imgur.com/pRnRHl8h.png)
```

File: keyboards/sawnsprojects/okayu/stm32f072/info.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "diode_direction": "COL2ROW",
+    "indicators": {
+        "caps_lock": "B12"
+    },
+    "ws2812": {
+        "pin": "B15",
+        "driver": "spi"
+    },
+        "rgblight": {
```
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f072/info.json:16]:
```suggestion
    "rgblight": {
```

File: keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#define STM32_PWM_USE_TIM3 TRUE
+
+#undef STM32_SPI_USE_SPI1
+#define STM32_SPI_USE_SPI1 TRUE
+
+
+#define WS2812_SPI SPID1 // default: SPID1
+#define WS2812_SPI_MOSI_PAL_MODE 5 // MOSI pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
+#define WS2812_SPI_SCK_PIN B3 // Required for F072, may be for others -- SCK pin, see the respective datasheet for the appropriate values for your MCU. default: unspecified
+#define WS2812_SPI_SCK_PAL_MODE 5 // SCK pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
```
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h:39]: This should be in your config.h, not here.

File: keyboards/sawnsprojects/okayu/stm32f303/chconf.h
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f303/chconf.h:1]: Is this actually needed?

File: keyboards/sawnsprojects/okayu/stm32f072/chconf.h
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f072/chconf.h:1]: Is this actually needed?

File: keyboards/sawnsprojects/okayu/stm32f303/info.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "diode_direction": "COL2ROW",
+    "indicators": {
+        "caps_lock": "B12"
+    },
+    "ws2812": {
+        "pin": "B15",
+        "driver": "spi"
+    },
+        "rgblight": {
```
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f303/info.json:16]:
```suggestion
    "rgblight": {
```

File: keyboards/sawnsprojects/okayu/stm32f303/mcuconf.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+#define STM32_PWM_USE_TIM3 TRUE
+
+#undef STM32_SPI_USE_SPI1
+#define STM32_SPI_USE_SPI1 TRUE
+
+
+#define WS2812_SPI SPID1 // default: SPID1
+#define WS2812_SPI_MOSI_PAL_MODE 5 // MOSI pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
+#define WS2812_SPI_SCK_PIN B3 // Required for F072, may be for others -- SCK pin, see the respective datasheet for the appropriate values for your MCU. default: unspecified
+#define WS2812_SPI_SCK_PAL_MODE 5 // SCK pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
```
Comment (drashna) [inline keyboards/sawnsprojects/okayu/stm32f303/mcuconf.h:39]: This needs to be in the config.h, not here.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/okayu/info.json [hunk 6]
```diff
@@ -0,0 +1,91 @@
+{
+    "keyboard_name": "Salane | Okayu",
```
Comment (dunk2k) [inline keyboards/sawnsprojects/okayu/info.json:2]:
```suggestion
    "keyboard_name": "Okayu",
```

File: keyboards/sawnsprojects/okayu/info.json
Comment (dunk2k) [inline keyboards/sawnsprojects/okayu/info.json:1]: @MaiTheSan 
Could you review 60% (60_*) layouts in `/layouts/default` for matching key arrangements this PCB matches.

Per match:
1. Append `"layouts"` with matching layout name and layout information, assign matrix positions/co-ordinates
2. Add `"community_layouts": []` to `info.json`, entering matching layout names (minus `LAYOUT_`) as comma separated JSON strings.
e.g. `"community_layouts" : ["60_ansi", "60_iso"]`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/okayu/info.json [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+        "pid": "0x00A1",
+        "device_version": "1.0.0",
+        "force_nkro": true
+    },
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": true,
+        "command": true,
```
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/info.json:17]: Unless there's a specific reason to have these enabled.
```suggestion
        "console": false,
        "command": false,
```

File: keyboards/sawnsprojects/okayu/keymaps/default/readme.md
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/sawnsprojects/okayu/keymaps/via/readme.md
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/sawnsprojects/okayu/stm32f303/info.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        }
+    }
```
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f303/info.json:31]:
```suggestion
```

File: keyboards/sawnsprojects/okayu/stm32f072/info.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        }
+    }
```
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f072/info.json:31]:
```suggestion
```

File: keyboards/sawnsprojects/okayu/info.json [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+        "nkro": true,
+        "rgblight": true
+    },
```
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/info.json:20]:
```suggestion
    },
    "bootloader": "stm32-dfu",
    "matrix_pins": {
        "cols": ["A9", "A0", "A8", "C13", "C4", "C2", "B11", "B10", "B1", "B0", "A7", "A6", "A5", "A4"],
        "rows": ["B3", "A15", "A3", "A10", "B14"]
    },
    "diode_direction": "COL2ROW",
    "indicators": {
        "caps_lock": "B12"
    },
    "ws2812": {
        "pin": "B15",
        "driver": "spi"
    },
    "rgblight": {
        "led_count": 24,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
    },
```

File: keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/handwired/onekey/stm32f0_disco/mcuconf.h -r platforms/chibios/GENERIC_STM32_F072XB/configs/mcuconf.h`
+ */
```
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h:20]:
```suggestion
// Copyright 2023 QMK
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f072/mcuconf.h:1]: This file should be moved to the top keyboard level (`keyboards/sawnsprojects/okayu`), and removed from here.

File: keyboards/sawnsprojects/okayu/stm32f303/mcuconf.h
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f303/mcuconf.h:1]: This file should be removed with the above suggestion.

File: keyboards/sawnsprojects/okayu/stm32f072/halconf.h
Comment (waffle87) [inline keyboards/sawnsprojects/okayu/stm32f072/halconf.h:1]: This file should be moved to the top keyboard level (`keyboards/sawnsprojects/okayu`), and removed from here.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/okayu/stm32f072/chconf.h
Comment (waffle87) [reply keyboards/sawnsprojects/okayu/stm32f072/chconf.h:1]: Then the file should be removed.

**Pull request closed without merging**